semgrep-core/tests/OTHER/rules/df_scala_expr_block.scala

```diff
@@ -0,0 +1,27 @@
+// https://linear.app/r2c/issue/PA-947
+
+package controllers
+
+import play.api.mvc._
+import javax.inject._
+import models.UserModel
+
+@Singleton
+class AuthController @Inject()(val controllerComponents: ControllerComponents) extends BaseController{
+  
+  def test1(input: String) = Action { implicit request: Request[AnyContent] =>
+    // ruleid: test
+    Ok(input)
+  }
+
+  def test2(input: String) = Action {
```

Comment:
Those Action { <stuff>; Ok(input) }  are represented in the generic AST as a StmtExpr (Block) ?
Where is Action in the AST then?

Those blocks are actually lambdas without any parameter I think in Scala.

Reply:
Yep indeed, they are represented as:

```
Call(
    N(Id(("Action", ...), ...)),
    [Arg(
        StmtExpr(
        Block(
            [ExprStmt(
            Call(
                N(
                Id(("Ok", ...), ...)),
                [Arg(
                    N(
                    Id(("input", ...), ...
                        )))]), ())])))])
```

I had a similar thought, that it could be fixed also by changing the translation. But I was not sure about the actual semantics in Scala, and anyways we had to handle `StmtExpr` at some point.

Reply:
ok sounds good!